File: php/src/Model/ColumnDef.php
<?php

declare(strict_types=1);

namespace Dokmatiq\DocGen\Model;

use Dokmatiq\DocGen\Internal\Serializer;

/** Column definition for table data. */
final class ColumnDef
{
    public function __construct(
        public readonly string $header,
        public readonly ?int $width = null,
        public readonly ?TextAlignment $alignment = null,
    ) {}

    /** @return array<string, mixed> */
    public function toArray(): array
    {
        return Serializer::toArray($this);
    }
}

File: java/src/main/java/com/dokmatiq/docgen/model/ColumnDef.java
package com.dokmatiq.docgen.model;

import com.fasterxml.jackson.annotation.JsonInclude;

/** Column definition for table data. */
@JsonInclude(JsonInclude.Include.NON_NULL)
public record ColumnDef(
        String header,
        Integer width,
        TextAlignment alignment,
        ColumnFormat format
) {
    public ColumnDef(String header, int width) {
        this(header, width, null, null);
    }

    public ColumnDef(String header, int width, TextAlignment alignment) {
        this(header, width, alignment, null);
    }
}
